### ISSUE TITLE
Emulation of Fairlight 2 (128k) fails on 128K Spectrum

### ISSUE THREAD

**Issue by mjwilson** [floooh/chips#90]
Going to https://floooh.github.io/tiny8bit/zx-ui.html?type=zx128 and trying to load a 128K snapshot of Fairlight 2, the emulator behaves incorrectly (usually the emulated Spectrum crashes, although the exact way in which this happens seems inconsistent).

[fairlight2.zip](https://github.com/floooh/chips/files/13853971/fairlight2.zip)

Expected behaviour: the player can move around the screen with Q/S/O/K. If they press O to walk up to the door, a message "LOCKED" appears. If they press 5 to select the key and walk to the door, they can enter the next room.

Actual behaviour: unexpected sounds from the beeper. If the player approaches the door then the emulated Spectrum locks up or crashes.

**Comment by floooh** [comment]
Yes that's not unexpected. The ZX Spectrum emulation by far isn't as good as the CPC, C64 and KC85 emulation. There's no memory contention for video memory access emulated at all for instance.
